### DISCUSSION TITLE
What's the difference between aliases and abstract commands?

### DISCUSSION THREAD

**Discussion by ghost** [houmain/keymapper#201] [Q&A] [answered]
It seems like they just do the same thing, but an alias defintion is more readable since it's in the form of `ActionName = Shortuct `, vs `Shortcut -> ActionName`, and furthermore the alias can just remain unused without affecting the behavior of the shortcut.

What's the actual use case for asbstract commands?

**Comment by ristomatti** [comment]
We had a lenghtly discussion on this on issue #148, but I think you can better grasp the difference from how they are used in the example config:
https://github.com/houmain/keymapper/blob/main/keymapper.conf

Abtract commands are context aware, which makes it possible to define a global mapping like `Alt{Comma} >> find`, then define what keys `find` will send based on application / OS.

**Reply by ghost** [reply]
Yeah I read that but couldn't see any difference. The alias notation seemingly does the same things as far as the user is concerned. And in both cases, nothing will happen to the layout if we remove the remapping in `[context]` below, besides the abstract command notation throwing a warning.

```
find = Alt{Comma}

[context]
find >> Control{Comma}
```

**Reply by houmain** [reply, marked as answer]
>  seemingly does the same things as far as the user is concerned

Yeah, one may get by with aliases alone.
Beside the validation that everything is properly mapped, these features come into my mind:

With abstract commands one only has to worry about the mapping order when the input side is defined:
```
Control{A} >> find
A >> findNext

[contextA]
findNext >> C
find >> D

[contextB]
find >> E
findNext >> F
```

They also allow to map multiple inputs to the same output:
```
A >> find
B >> find
find >> C
```
